pkg/kotsadmconfig/validation/validator.go

```diff
@@ -0,0 +1,68 @@
+package validation
+
+import (
+	"fmt"
+	"regexp"
+
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+)
+
+const (
+	BoolItemType      = "bool"
+	FileItemType      = "file"
+	TextItemType      = "text"
+	LabelItemType     = "label"
+	HeadingItemType   = "heading"
+	PasswordItemType  = "password"
+	TextAreaItemType  = "textarea"
+	SelectOneItemType = "select_one"
+)
+
+const (
+	regexMatchError = "Value does not match regex"
+)
+
+var (
+	validatableItemTypesMap = map[string]bool{
+		TextItemType:     true,
+		PasswordItemType: true,
+		TextAreaItemType: true,
+		FileItemType:     true,
+	}
+)
+
+func isValidatableConfigItem(item kotsv1beta1.ConfigItem) bool {
+	return item.Validation != nil && !item.Hidden && item.When != "false" && !item.Repeatable && validatableItemTypesMap[item.Type]
+}
+
+func validate(value string, validator kotsv1beta1.ConfigItemValidator) []ValidationError {
+	var validationErrs []ValidationError
+
+	if validator.Regex != nil {
+		validationErr := validateRegex(value, validator.Regex)
+		if validationErr != nil {
+			validationErrs = append(validationErrs, *validationErr)
+		}
+	}
+
+	return validationErrs
+}
+
+func validateRegex(value string, regexValidator *kotsv1beta1.RegexValidator) *ValidationError {
+	regex, err := regexp.Compile(regexValidator.Pattern)
+	if err != nil {
+		return &ValidationError{
+			ValidationErrorMessage: fmt.Sprintf("Invalid regex: %s", err.Error()),
```

Comment:
Why isn't this:
```suggestion
			ValidationErrorMessage: regexValidator.Message,
```

Reply:
vendor's message will be very ambiguous.
adding another message field controlled by us would be more accurate.
for regex, there can be 2 errors
- when value doesn't match regex: `value does not match regex` --> easier to know the string was not matched
- and, when regex itself is invalid: Invalid regex: <<error from the regex lib>>
eg: when Pattern: "["  err would be:
`"Invalid regex: error parsing regexp: missing closing ]: `[`"` --> makes easier for debugging the validation err